src/validation.cpp

```diff
@@ -2650,6 +2649,13 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
     const auto consensus = chainparams.GetConsensus();
     auto isEvmEnabledForBlock = IsEVMEnabled(pindex->nHeight, mnview, consensus);
 
+    uint64_t evmQueueId{};
+    if (isEvmEnabledForBlock) {
+        auto r = XResultValueLogged(evm_try_unsafe_create_queue(result, pindex->GetBlockTime()));
```

Comment:
With queue creation moved to inside ConnectBlock from ConnectTip how do we delete the queue if ConnectBlock returns early with a failure? We may end up with dangling queues.

Reply:
These changes are in PR #2521. Let me shift these into that PR instead.